src/AnimIfChange.jsx

```diff
@@ -1,11 +1,96 @@
 'use strict';
 
-var React = require('react');
-var CssAnimation = require('css-animation');
+import React, {Component} from 'react';
+import anim from 'css-animation';
+const {findDOMNode, cloneElement, createElement} = React;
+const ref = '__el';
+
+function isEmpty(obj) {
+  return obj === null || obj === undefined;
+}
+
+export default class AnimIfChange extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      children: this.props.children
+    };
+    this.requestRemoveChildren = this.requestRemoveChildren.bind(this);
+  }
+
+  renderContainer(children) {
+    return createElement(
+      this.props.container,
+      null,
+      children
+    );
+  }
+
+  getChildrenToRender(component) {
+    return this.state.children && cloneElement(React.Children.only(component), {ref: ref});
+  }
 
-class AnimIfChange extends React.Component {
   render() {
-    return React.Children.only(this.props.children);
+    const child = this.getChildrenToRender(this.state.children);
+    return this.renderContainer(child);
+  }
+
+  componentDidMount() {
+    this.componentDidUpdate();
+  }
+
+  anim(el, transitionName, enter, fn) {
+    if (transitionName) {
+      anim(el, transitionName + (enter ? '-enter' : '-leave'), fn);
+    } else if (fn) {
+      fn();
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({children: nextProps.children});
+  }
+
+  requestRemoveChildren(removeRequired) {
+    return () => {
+      if (removeRequired) {
+        this.setState({children: null});
+      }
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    const props = this.props;
+    const domNode = findDOMNode(this.refs[ref]);
+    if (!domNode) {
+      return;
+    }
+    // render at first time
+    if (isEmpty(prevProps)) {
+      if (props.defaultTransitionEnter) {
```

Comment:
if (props.defaultTransitionEnter !== props.transitionEnter){
  this.anim(domNode, props.transitionName, props.transitionEnter);
}